src/TLuaInterpreter.cpp

```diff
@@ -373,35 +373,39 @@ int TLuaInterpreter::raiseEvent(lua_State* L)
     TEvent event;
 
     int n = lua_gettop(L);
-    for (int i = 1; i <= n; i++) {
-        if (lua_isnumber(L, 1)) {
-            event.mArgumentList.append(QString::number(lua_tonumber(L, 1)));
-            event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
-            lua_remove(L, 1);
-        } else if (lua_isstring(L, 1)) {
-            event.mArgumentList.append(QString::fromUtf8(lua_tostring(L, 1)));
-            event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
-            lua_remove(L, 1);
-        } else if (lua_isboolean(L, 1)) {
-            event.mArgumentList.append(QString::number(lua_toboolean(L, 1)));
-            event.mArgumentTypeList.append(ARGUMENT_TYPE_BOOLEAN);
-            lua_remove(L, 1);
-        } else if (lua_isnil(L, 1)) {
-            event.mArgumentList.append(QString());
-            event.mArgumentTypeList.append(ARGUMENT_TYPE_NIL);
-            lua_remove(L, 1);
-        } else if (lua_istable(L, 1)) {
-            event.mArgumentList.append(QString::number(luaL_ref(L, LUA_REGISTRYINDEX)));
-            event.mArgumentTypeList.append(ARGUMENT_TYPE_TABLE);
-        } else if (lua_isfunction(L, 1)) {
-            event.mArgumentList.append(QString::number(luaL_ref(L, LUA_REGISTRYINDEX)));
-            event.mArgumentTypeList.append(ARGUMENT_TYPE_FUNCTION);
+    // We go from the top of the stack down, because luaL_ref will
+    // only reference the object at the top of the stack
+    for (int i = n; i >= 1; i--) {
```

Comment:
You don't really need to count backwards, so I am not sure what is more readable, a positive or negative for loop. I'll leave that open for you.

Reply:
I'm not sure how to do it without counting backwards. `luaL_ref` doesn't take an indexing argument, it just takes the top object on the stack. So either I'd have to iterate through the stack from the top down, popping as I go, or if I count forwards, I'd have to copy the object to the top of the stack, call  `luaL_ref`, then pop it off. I could do the second option, I guess, it just feels weird to manipulate the stack while parsing it. Could bring up some issues.

Reply:
If I am not mistaken, you never really use the `i` variable within the loop. All you need it for is to count the number of loops you made already.

Reply:
Hrm... its used for error reporting, when there's a bad argument type, but that could be replaced with a call to `lua_gettop()` I think.

I'll change it if it's requested, but I do kind of like the reinforcement it gives that we're going top down thru the stack. You're right though, it isn't necessary.

Reply:
As I said, I'll leave it to you. I usually think in forward loops and need to doubletake on backwards, which is why I noticed that. ;)